### PR TITLE
Issue:257 Fix Bulk lookup FIB for IPV4

### PR DESCRIPTION
**Pull request by vikasJain85**
Issue: https://github.com/CloudNativeDataPlane/cndp/issues/257
Signed-off-by: Vikas Jain [vikas.jain@intel.com](mailto:vikas.jain@intel.com)
Signed-off-by: Vikas Jain<vikas.jain@intel.com>

### CODE REVIEW

**Comment by vikasJain85** [comment]
Issue: https://github.com/CloudNativeDataPlane/cndp/issues/257
Signed-off-by: Vikas Jain [vikas.jain@intel.com](mailto:vikas.jain@intel.com)

**Comment by vikasJain85** [comment]
retest

**Comment by KeithWiles** [comment]
To help fix the DCO error, click on the `Details` link next to the DCO action to get some help with adding the Signed-Off-by line.

File: examples/cndpfwd/main.c [hunk 1]
```diff
@@ -160,24 +158,36 @@ _l3fwd_test(jcfg_lport_t *lport, struct fwd_info *fwd)
     if (n_pkts == PKTDEV_ADMIN_STATE_DOWN)
         return -1;
 
+    struct ether_addr *eaddr = malloc (n_pkts * sizeof(struct ether_addr));
+    uint16_t *tx_port = malloc (n_pkts * sizeof(uint16_t));
+    uint32_t *ip_addr = malloc (n_pkts * sizeof(uint32_t));
+
+    if (eaddr == NULL || tx_port == NULL || ip_addr) {
```
Comment (KeithWiles) [inline examples/cndpfwd/main.c:165]: The coding style for CNDP does not allow for single line for, if, while, do, ... statements to have {} around the line.

The CONTRIBUTING.md file should help with coding style.

**Review by KeithWiles — CHANGES_REQUESTED** [review]
Some minor comments to style, but I will be reviewing this code more later.

**Review by jeffreybshaw — CHANGES_REQUESTED** [review]

File: examples/cndpfwd/l3-fwd.c [hunk 2]
```diff
@@ -76,13 +76,20 @@ l3fwd_fib_populate(struct fwd_info *fwd, struct cne_fib *fib)
 }
 
 int
-l3fwd_fib_lookup(uint32_t *ip, struct ether_addr *eaddr, uint16_t *tx_port)
+l3fwd_fib_lookup(uint32_t *ip, struct ether_addr *eaddr, uint16_t *tx_port, int n)
 {
-    uint64_t nexthop;
+    uint64_t *nexthop = malloc(n * sizeof(uint64_t));
```
[the remark this answers is not in the document]
Comment (KeithWiles) [reply examples/cndpfwd/l3-fwd.c:81]: The malloc() also has a memory leak as nexthop is not freed.

How about changing this to:

``` c
uint16_t nhop[MAX_BURST];
uint16_t *nexthop = &nhop[0];
```

File: examples/cndpfwd/main.c [hunk 3]
```diff
@@ -160,24 +158,34 @@ _l3fwd_test(jcfg_lport_t *lport, struct fwd_info *fwd)
     if (n_pkts == PKTDEV_ADMIN_STATE_DOWN)
         return -1;
 
+    struct ether_addr *eaddr = malloc (n_pkts * sizeof(struct ether_addr));
```
[the remark this answers is not in the document]
Comment (KeithWiles) [reply examples/cndpfwd/main.c:161]: Same here:

``` c
struct ether_addr ethaddr[MAX_BURST];
struct ether_addr *eaddr = &ethaddr[0];
```
```

File: examples/cndpfwd/main.c [hunk 4]
```diff
@@ -160,24 +158,34 @@ _l3fwd_test(jcfg_lport_t *lport, struct fwd_info *fwd)
...
+    uint16_t *tx_port = malloc (n_pkts * sizeof(uint16_t));
```
Comment (KeithWiles) [inline examples/cndpfwd/main.c:162]:
``` c
uint16_t tport[n_pkts];
uint16_t *tx_port = &tport[0];
```

File: examples/cndpfwd/main.c [hunk 5]
```diff
@@ -160,24 +158,34 @@ _l3fwd_test(jcfg_lport_t *lport, struct fwd_info *fwd)
...
+    uint32_t *ip_addr = malloc (n_pkts * sizeof(uint32_t));
```
Comment (KeithWiles) [inline examples/cndpfwd/main.c:163]:
``` c
uint32_t ipaddrs[n_pkts];
uint32_t *ip_addr = &ipaddrs[0];
```

**Review by KeithWiles — APPROVED** [review]

**Pull request merged**